w3-account.md

```diff
@@ -0,0 +1,224 @@
+
+
+# Terminology
+
+## Account
+
+User account is a [principal][] identified by [`did:mailto`][] (listed in a UCAN's `iss` or `aud` field).
+
+When user agent creates a new space, it MAY delegate full or subset of the capabilities to user account so they could be reclaimed by a user with another agent.
+
+> Example shows all capabilities to `did:key:zAlice` space been delegated to the user account `alice@web.mail` 
+
+```ts!
+{
+  iss: "did:key:zAlice",
+  aud: "did:mailto:alice@web.mail",
+  att: [{ with: "did:key:zAlice", can: "*" }],
+  exp: null,
+  sig: "..."
+}
+```
+
+## Delegations
+
+Agent MAY publish account delegations to a serivce so that they can be persisted and retrieved later with a different agent.
+
+> Invokes `access/delegate` asking web3.storage to record delegation from `did:key:zAlice` space to the `alice@web.mail` account.
+
+ ```ts
+{
+  iss: "did:key:zAgent",
+  aud: "did:dns:web3.storage",
+  att: [{
+    with: "did:key:zAgent",
+    can: "access/delegate",
+    nb: {
+      access: Link<{
+        iss: "did:key:zAlice",
+        aud: "did:mailto:alice@web.mail",
+        att: [{ with: "did:key:zAlice", can: "*" }],
+        exp: null,
+        sig: "..."
+      }>
+     }
+    }
+  }],
+  sig: "..."
+}
+```
+
+:::warning
+[Recipient validation][] requires wrapping actual delegation into `access/delegate` invocation, but in the future we may find a way to remove this requirement.
+:::
+ 
+## Authorization
+
+User MAY authorize an agent to represent their account by invoking `ucan/issue` capability with an audience where authorization is to be considered valid
+
+
+> Agent requests authorization to represent `alice@web.mail` with `did:key:zAgent` in the context of `web3.storage`
+
+```ts
+{
+  iss: "did:key:zAgent",
+  aud: "did:dns:web3.storage",
+  att: [{
+    with: "did:key:zAgent",
+    can: "ucan/issue",
+    nb: { as: "did:mailto:alice@web.mail" }
+  }]
+}
+```
+
+### issue `with`
+
+Resource MUST be a [`did:key`][] URI. It represents a public key that user wishes to use for signing UCANs issued by DID in the [`nb.as`][issue `as`] field.
+
+### issue `as`
+
+Account MUST be a [`did:mailto`][] principal. It is an account that user wishes to be represented by [`did:key`][] in the [`with`][issue `with`] field.
+
+
+## Verification
+
+Service MUST perform out of bound user verification e.g. send a link to the mailbox of the account by clicking which user would authorize agent to represent the account.
+
+On succesful verification service MUST delegate corresponding `ucan/sign` capability to the [`did:key`][] it was requested [`with`][issue `with`].
+
+Delegation represents authorization to issue [UCAN][]s with [`did:mailto`] account prinicipal which MAY be signed with [`did:key`] of the agent.
+
+
+```ts
+{
+  iss: "did:dns:web3.storage",
+  aud: "did:mailto:alice@web.mail",
+  att: [{
+    with: "did:dns:web3.storage",
+    can: "ucan/sign",
+    nb: { as: "did:key:zAgent" }
+  }],
+  exp: null
+  sig: "..."
+}
```

Comment:
```js
{
iss: "did:dns:web3.storage",
aud: "...",
att: [{
  can: "send/message",
  with: "did:dns:web3.storage",
}],
prf: [{
  iss: "did:key:root",
  aud: "did:dns:web3.storage",
  att: [{
    with: "did:dns:web3.storage",
    can: "ucan/sign",
    nb: { as: "did:key:zAgent" }
  }],
  exp: null
  sig: "..."
}]
```

Reply:
@expede can you please provide feedback on this

Reply:
@Gozala wrt `send/message`, that's new right? I notice that it is of the form `verb/noun` whereas most of the time our caps name convention is more like 'noun/verb'.

Reply:
Hey @expede have you had a chance to consider this ? In our call, I believe you said you were ok with general idea, yet wanted to ponder more about actual representation

Reply:
Sorry for the post-merge feedback @Gozala !

> wrt send/message, that's new right? I notice that it is of the form verb/noun whereas most of the time our caps name convention is more like noun/verb

@gobengo I've been thinking about this in a UCAN "standard library" as `msg/send`, which cuts across email, text messaging, message passing, etc etc.


> ```
> iss: "did:dns:web3.storage", // <- Maybe this should be did:key:root here
> aud: "did:dns:web3.storage",
> ```

I continue to find `iss == aud` deeply strange. I agree with your comment "Maybe this should be did:key:root here".

Rather than granting the ability to sign "as" another DID, I think it's cleaner to make this a "forwarding edge", which follows much more the capabilities worldview. I _think_ that the end result is the same, so it's just naming.

```
Time 1: carol --[fwd_all]--> dan
Time 2: alice --[can: foo]--> bob --[can: foo]--> carol
Time 3: dan --[can: foo]--> frank
```

BECAUSE we can substitute a link:

```
alice --[can: foo]--> bob --[can: foo]--> carol
                                            |
                                        [fwd_all]
                                            |
                                            V
                                           dan --[can: foo]--> frank

End result: Frank can use Alice's foo
```

This is _kind of_ like "signing as" Carol, but it's only at the level of authority, not identity. _Alice_ saying "you can sign as Bob" is going to get really confusing unless you want to do DID management, which IMO should be at a DID layer. I think you can manage all of this cleanly at the layer of capabilities.

This is a powerful enough feature that has enough uses that I propose that we add this as a first-class feature to UCAN at some stage. In the meantime, I propose that we call this something other than "signing as".

Here's some delegations in this model:

```
w3s --[fwd_all]--> didJune
w3s --[fwd_all]--> didJuly
w3s --[fwd_all]--> didAug

didJune --[w3s's foo]--> alice

bob --[bob's bar]--> w3s
                      |
                   [fwd_all]
                      |
                      V
                   didJune --[bob's bar]--> carol
```

Here's that last one concretely (well, the `bob -> w3s -> june` portion just to keep it small _enough_ for an example)

```js
{
  iss: "did:key:june",
  aud: "did:key:carol",
  att: [{ can: "thing/doSomthing", with: "resource://example.com" }],
  sig: "...",
  prf: [
    {
      iss: "did:key:zBob",
      aud: "did:dns:web3.storage",
      att: [{
        with: "resource://exmaple.com",
        can: "thing/doSomething"
      }],
    exp: august2022,
    sig: "..."
  },
  {
      iss: "did:dns:web3.storage",
      aud: "did:key:june",
      att: [{
        with: "did:dns:web3.storage",
        can: "ucan/forward_all" // we can come up with a better label
      }],
      exp: november2022,
      sig: "..."
    }
  ]
}
```